### PR TITLE
subsys: random: xoroshiro128: Use entropy get isr during init

### PR DESCRIPTION
**Pull request by AdithyaBaglody**
Current implementation of the xoroshiro depends on the ISR being
triggered when the interrupts is locked. This patch proposes
implementing the init with entropy_get_isr. This implementation
can be called at PRE_KERNEL_2 stage, even when the interrupts
are locked.

Fixes: #8199

Signed-off-by: Adithya Baglody <adithya.nagaraj.baglody@intel.com>

### CODE REVIEW

File: subsys/random/rand32_xoroshiro128.c
Comment (carlescufi) [inline subsys/random/rand32_xoroshiro128.c]: I would do it in the same way the kernel does: https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/init.c#L406, by checking specifically for `ENOTSUP`.

Comment (carlescufi) [inline subsys/random/rand32_xoroshiro128.c]: Doesn't more fit in the same line before you hit the 80 char width?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: subsys/random/rand32_xoroshiro128.c
Comment (AdithyaBaglody) [reply subsys/random/rand32_xoroshiro128.c]: sure. I fixed that.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@AdithyaBaglody thank you! approved

**Comment by carlescufi** [comment]
@nashif we can merge this unless you want an additional review from @lpereira

**Review by nashif — APPROVED** [review]
looks good, but will wait for @lpereira

**Review by lpereira — APPROVED** [review]
LGTM.

**Pull request merged**
